src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -132,8 +139,12 @@ int TPCHandler::opensocket_callback(void *clientp,
   {XrdNetAddr thePeer(&(aInfo->addr));
     rec->isIPv6 =  (thePeer.isIPType(XrdNetAddrInfo::IPv6)
                     && !thePeer.isMapped());
-    // Register the socket to the packet marking manager
-    rec->pmarkManager.addFd(fd,&aInfo->addr);
+    std::stringstream connectErrMsg;
+
+    if(!rec->pmarkManager.connect(fd,&(aInfo->addr),aInfo->addrlen,5,connectErrMsg)) {
```

Comment:
You are correct, hardcoding this may become problematic. The Linux default is 60 seconds which some people claim is too long. It all depends on what you are trying to accomplish. So, yes, I would make this a configurable number. Note that as far as I can tell, curl will use the Linux default. So, it is somewhat strange that when scitag is not enabled it's a long timeout and when it is enabled it is a very short timeout. That will be hard to explain. I would make sure both values are the same.

Reply:
OK, will try to see how I can extract that default value from curl and assign the same value for it...

Reply:
OK, there's no way I can extract this value. The default is 300 seconds: https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT.html

 Though I can add a static variable and set the timeout value of 300 seconds with a reference to that documentation

Reply:
I don't understand. The Linux default is 60 seconds the curl default is 300 second? I don't think so. I mean 5 minutes is totally excessive. Can't we just set the curl timeout irrespective of whether pmark is enabled or not to the Linux default of 60 seconds?

Reply:
I am indeed more than happy to set the value for 60 seconds for curl, even if pmark is not enabled.

Reply:
Done, if everything is good then I can squash and force push before merging.

@abh3 out of curiosity, where did you get the information about the Linux default 60 seconds connect timeout value?